### DISCUSSION TITLE
How to improve performance on production

### DISCUSSION THREAD

**Discussion by joaovdiasb** [filamentphp/filament#4493] [Help] [answered]
On production server takes about 4s to conditional rendering with hidden(), on bigger forms can go to 14s, tried with nginx and apache, approximate results.

<img width="540" alt="CleanShot 2022-10-11 at 21 48 25@2x" src="https://user-images.githubusercontent.com/51106311/195223965-a1512ab2-9d11-40b3-82b0-e47cfa75b5c0.png">

[code](https://github.com/joaovdiasb/fillament-process/blob/master/app/Filament/Resources/AttachmentTypeResource/Forms/AttachmentTypeForm.php)

on bigger form
<img width="543" alt="CleanShot 2022-10-11 at 21 49 14@2x" src="https://user-images.githubusercontent.com/51106311/195224030-538578f0-7635-4a69-b12a-aed9bd0d555a.png">
[code](https://github.com/joaovdiasb/fillament-process/blob/master/app/Filament/Resources/ProjectApprovalResource/Forms/ProjectForm.php)

**Comment by danharrin** [comment, marked as answer]
Install debugbar.

How long is the actual server response?

How much memory is used?

How many views are used?

How long is your application boot time?

How long do your database queries take?

Also, is opcache enabled?

**Reply by joaovdiasb** [reply]
> Install debugbar.
> 
> How long is the actual server response?
> 
> How much memory is used?
> 
> How many views are used?
> 
> How long is your application boot time?
> 
> How long do your database queries take?
> 
> Also, is opcache enabled?

Thanks again @danharrin
The server specs was handling fine, opcache installed but without aditional configs, this solved the problem:
```
opcache.enable=1
opcache.memory_consumption=128
opcache.max_accelerated_files=10000
opcache.revalidate_freq=200
```

and thanks for the library, it's beautiful
